src/ResourceManagement/EventHub/AuthorizationRuleBaseImpl.cs

```diff
@@ -0,0 +1,136 @@
+
+        ///GENMHASH:AB1BA95F78B711F10D574A0046DE17B7:D03B292D619886851FBEC2556FBA2D32
+        public IReadOnlyList<AccessRights> Rights
+        {
+            get
+            {
+                if (this.Inner.Rights == null)
+                {
+                    return new List<AccessRights>();
+                }
+                else
+                {
+                    List<AccessRights> rights = new List<AccessRights>();
+                    foreach (var r in this.Inner.Rights) 
+                    {
+                        rights.Add(AccessRights.Parse(r));
+                    }
+                    return rights;
+                }
+            }
+        }
+
+        ///GENMHASH:B5F20FEE4712239FEC489FA348B7432B:8231BB581A68B1C17FCEAAEEBC857049
+        public RuleImpl WithListeningEnabled()
+        {
+            if (this.Inner.Rights == null)
+            {
+                this.Inner.Rights = new List<string>();
+            }
+            if (!this.Inner.Rights.Contains("Listen"))
+            {
+                this.Inner.Rights.Add("Listen");
+            }
+            return this as RuleImpl;
+        }
+
+        ///GENMHASH:C5DDA67F1816E477A8EAA6ECEFBBB25C:A23796BF710470D2B0F0EF3CB21F9D0B
+        public RuleImpl WithSendingEnabled()
+        {
+            if (this.Inner.Rights == null)
+            {
+                this.Inner.Rights = new List<string>();
+            }
+            if (!this.Inner.Rights.Contains("Send"))
+            {
+                this.Inner.Rights.Add("Send");
+            }
+            return this as RuleImpl;
+        }
+
+        ///GENMHASH:D56754248D4EE259AEA7A819BD939780:448F969B7719EB9B90B397331F814EA9
+        public RuleImpl WithManagementEnabled()
+        {
+            WithListeningEnabled();
+            WithSendingEnabled();
+            if (this.Inner.Rights == null)
+            {
+                this.Inner.Rights = new List<string>();
+            }
+            if (!this.Inner.Rights.Contains("Manage"))
+            {
+                this.Inner.Rights.Add("Manage");
+            }
+            return this as RuleImpl;
+        }
+
+
+        ///GENMHASH:323E13EA523CC5C9992A3C5081E83085:27E486AB74A10242FF421C0798DDC450
+        protected abstract Task<AccessKeysInner> GetKeysInnerAsync(CancellationToken cancellationToken = default(CancellationToken));
+
+        ///GENMHASH:2751D8683222AD34691166D915065302:9B86FC2DF5BE701D70A2B1F445F91E61
+        public async Task<Microsoft.Azure.Management.Eventhub.Fluent.IEventHubAuthorizationKey> GetKeysAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var keysInner = await this.GetKeysInnerAsync(cancellationToken);
+            return new EventHubAuthorizationKeyImpl(keysInner);
+        }
+
+        ///GENMHASH:E4DFA7EA15F8324FB60C810D0C96D2FF:EACB47A475A2EC02AC952C3B5FCFCC6E
+        public IEventHubAuthorizationKey GetKeys()
+        {
+            return GetKeysAsync().Result;
```

Comment:
>GetKeysAsync().Result [](start = 19, length = 21)

do not call directly asyc.Result. Please use Extensions.Synchronize method instead. this one is must fix.